[REPO STATUS]
project build/make/                             (*** NO BRANCH ***)
 -m	core/Makefile
 -m	core/java.mk
 -m	core/product.mk
 -m	target/product/base_system.mk
 -m	target/product/full_base.mk
 -m	target/product/handheld_product.mk
 -m	target/product/handheld_system.mk
 -m	target/product/handheld_system_ext.mk
 -m	target/product/media_system.mk
 -m	target/product/runtime_libart.mk
 -m	target/product/telephony_system.mk

project build/soong/                            (*** NO BRANCH ***)
 -m	ui/build/finder.go

project external/noto-fonts/                    (*** NO BRANCH ***)
 -m	fonts.mk
 -m	notosanscjk/Android.bp
 --	notosanscjk/NotoSansCJK.ttc
 -m	notoserifcjk/Android.bp
 --	notoserifcjk/NotoSerifCJK-Bold.ttc
 -m	scripts/subset_noto_cjk.py

project external/vim/                           (*** NO BRANCH ***)
 -d	Android.mk

project frameworks/base/                        (*** NO BRANCH ***)
 -m	core/java/android/os/Build.java
 -m	data/fonts/fonts.xml

project frameworks/native/                      (*** NO BRANCH ***)
 -m	cmds/atrace/Android.bp

project packages/modules/adb/                   (*** NO BRANCH ***)
 -m	daemon/main.cpp
 -m	root/adbroot_service.cpp
 -m	sockets.cpp

project system/core/                            (*** NO BRANCH ***)
 -m	init/Android.bp
 -m	rootdir/init.rc

project system/sepolicy/                        (*** NO BRANCH ***)
 -m	build/soong/policy.go

project vendor/lineage/                         (*** NO BRANCH ***)
 -m	bootanimation/Android.mk
 -m	config/common.mk
 -m	config/common_full.mk
 -m	config/common_mobile.mk
 -m	config/telephony.mk
 -m	prebuilt/common/etc/init/init.lineage-system_ext.rc


[REPO DIFF]
project build/make/
diff --git a/core/Makefile b/core/Makefile
index 3ed8504..4af6395 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -6413,7 +6413,7 @@ INTERNAL_OTA_METADATA := $(PRODUCT_OUT)/ota_metadata
 $(call declare-0p-target,$(INTERNAL_OTA_METADATA))

 ifeq ($(TARGET_BUILD_VARIANT),user)
-    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
+    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
 else
 ifneq ($(LINEAGE_BUILD),)
     $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true

diff --git a/core/java.mk b/core/java.mk
index b13ef4d..69ccf8a 100644
--- a/core/java.mk
+++ b/core/java.mk
@@ -212,7 +212,7 @@ $(full_classes_compiled_jar): PRIVATE_WARNINGS_ENABLE := $(LOCAL_WARNINGS_ENABLE
 # available via JDWP.
 ifneq (,$(PRODUCT_MINIMIZE_JAVA_DEBUG_INFO))
 ifneq (,$(filter userdebug user,$(TARGET_BUILD_VARIANT)))
-LOCAL_JAVACFLAGS+= -g:source,lines
+LOCAL_JAVACFLAGS+= -g:none
 endif
 endif

diff --git a/core/product.mk b/core/product.mk
index 0938926..46afcbd 100644
--- a/core/product.mk
+++ b/core/product.mk
@@ -527,7 +527,12 @@ _readonly_late_variables += \
   PRODUCT_COPY_FILES \
   PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING \
   PRODUCT_DEX_PREOPT_BOOT_FLAGS \
-  PRODUCT_SOONG_NAMESPACES
+  PRODUCT_SOONG_NAMESPACES \
+  PRODUCT_DEX_PREOPT_DEFAULT_COMPILER_FILTER \
+  PRODUCT_DEX_PREOPT_DEFAULT_FLAGS \
+  PRODUCT_OTHER_JAVA_DEBUG_INFO \
+  PRODUCT_SYSTEM_SERVER_COMPILER_FILTER \
+  PRODUCT_SYSTEM_SERVER_DEBUG_INFO

 _readonly_early_variables := $(filter-out $(_readonly_late_variables),$(_product_var_list))

diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index ba239a2..38294f6 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -32,9 +32,7 @@ PRODUCT_PACKAGES += \
     appops \
     app_process \
     appwidget \
-    atrace \
     audioserver \
-    BackupRestoreConfirmation \
     bcc \
     blank_screen \
     blkid \
@@ -44,8 +42,6 @@ PRODUCT_PACKAGES += \
     boringssl_self_test \
     bpfloader \
     bu \
-    bugreport \
-    bugreportz \
     cgroups.json \
     charger \
     cmd \
@@ -79,9 +75,6 @@ PRODUCT_PACKAGES += \
     com.android.wifi \
     ContactsProvider \
     content \
-    CtsShimPrebuilt \
-    CtsShimPrivPrebuilt \
-    debuggerd\
     device_config \
     dmctl \
     dnsmasq \
@@ -91,7 +84,6 @@ PRODUCT_PACKAGES += \
     dump.erofs \
     dumpstate \
     dumpsys \
-    DynamicSystemInstallationService \
     e2fsck \
     ExtShared \
     flags_health_check \
@@ -107,7 +99,6 @@ PRODUCT_PACKAGES += \
     group_system \
     gsid \
     gsi_tool \
-    heapprofd \
     heapprofd_client \
     gatekeeperd \
     gpuservice \
@@ -278,9 +269,6 @@ PRODUCT_PACKAGES += \
     tc \
     telecom \
     telephony-common \
-    tombstoned \
-    traced \
-    traced_probes \
     tune2fs \
     uiautomator \
     uinput \
@@ -321,11 +309,6 @@ ifeq ($(EMMA_INSTRUMENT),true)
   endif # EMMA_INSTRUMENT_STATIC
 endif # EMMA_INSTRUMENT

-ifeq (,$(DISABLE_WALLPAPER_BACKUP))
-  PRODUCT_PACKAGES += \
-    WallpaperBackup
-endif
-
 # For testing purposes
 ifeq ($(FORCE_AUDIO_SILENT), true)
     PRODUCT_SYSTEM_PROPERTIES += ro.audio.silent=1
@@ -377,9 +360,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += system/core/rootdir/init.zygote32.rc:system/etc/init/hw/init.zygote32.rc
 PRODUCT_VENDOR_PROPERTIES += ro.zygote?=zygote32

-PRODUCT_SYSTEM_PROPERTIES += debug.atrace.tags.enableflags=0
-PRODUCT_SYSTEM_PROPERTIES += persist.traced.enable=1
-
 # Packages included only for eng or userdebug builds, previously debug tagged
 PRODUCT_PACKAGES_DEBUG := \
     adb_keys \
@@ -421,11 +401,6 @@ endif
 PRODUCT_SYSTEM_SERVER_APPS += \
     SettingsProvider \

-ifeq (,$(DISABLE_WALLPAPER_BACKUP))
-  PRODUCT_SYSTEM_SERVER_APPS += \
-    WallpaperBackup
-endif
-
 PRODUCT_PACKAGES_DEBUG_JAVA_COVERAGE := \
     libdumpcoverage

diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index 7aa7903..5f5aad1 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -25,7 +25,6 @@ PRODUCT_PACKAGES := \

 PRODUCT_PACKAGES += \
     LiveWallpapersPicker \
-    PhotoTable \
     preinstalled-packages-platform-full-base.xml

 # Net:

diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index b2ce4c5..c468988 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -31,7 +31,6 @@ PRODUCT_PACKAGES += \
     preinstalled-packages-platform-handheld-product.xml \
     QuickSearchBox \
     SettingsIntelligence \
-    frameworks-base-overlays

 ifeq ($(LINEAGE_BUILD),)
 PRODUCT_PACKAGES += \

diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index d9edca6..f3f3d0a 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -32,11 +32,8 @@ $(call inherit-product-if-exists, frameworks/base/data/keyboards/keyboards.mk)
 $(call inherit-product-if-exists, frameworks/webview/chromium/chromium.mk)

 PRODUCT_PACKAGES += \
-    BasicDreams \
     BlockedNumberProvider \
     BluetoothMidiService \
-    BookmarkProvider \
-    BuiltInPrintService \
     CalendarProvider \
     cameraserver \
     CameraExtensionsProxy \
@@ -46,7 +43,6 @@ PRODUCT_PACKAGES += \
     DeviceAsWebcam \
     DocumentsUI \
     DownloadProviderUi \
-    EasterEgg \
     ExternalStorageProvider \
     FusedLocation \
     InputDevices \
@@ -55,17 +51,13 @@ PRODUCT_PACKAGES += \
     ManagedProvisioning \
     MmsService \
     MtpService \
-    MusicFX \
     NfcNci \
     PacProcessor \
     preinstalled-packages-platform-handheld-system.xml \
-    PrintRecommendationService \
     PrintSpooler \
     ProxyHandler \
     screenrecord \
     SecureElement \
-    SharedStorageBackup \
-    SimAppDialog \
     Telecom \
     TelephonyProvider \
     TeleService \

diff --git a/target/product/handheld_system_ext.mk b/target/product/handheld_system_ext.mk
index 187b627..5d5f758 100644
--- a/target/product/handheld_system_ext.mk
+++ b/target/product/handheld_system_ext.mk
@@ -28,4 +28,3 @@ PRODUCT_PACKAGES += \
     Settings \
     StorageManager \
     SystemUI \
-    WallpaperCropper \

diff --git a/target/product/media_system.mk b/target/product/media_system.mk
index 94d9177..f51e20d 100644
--- a/target/product/media_system.mk
+++ b/target/product/media_system.mk
@@ -70,10 +70,6 @@ PRODUCT_VENDOR_PROPERTIES += \
     tombstoned.max_tombstone_count?=50
 endif

-PRODUCT_VENDOR_PROPERTIES += \
-    ro.logd.size.stats=64K \
-    log.tag.stats_log=I
-
 # Enable CFI for security-sensitive components
 $(call inherit-product, $(SRC_TARGET_DIR)/product/cfi-common.mk)
 $(call inherit-product-if-exists, vendor/google/products/cfi-vendor.mk)

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index 68ed249..32556aa 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -110,8 +110,8 @@ ifeq (eng,$(TARGET_BUILD_VARIANT))
         pm.dexopt.boot-after-ota?=extract
 else
     PRODUCT_SYSTEM_PROPERTIES += \
-        pm.dexopt.first-boot?=verify \
-        pm.dexopt.boot-after-ota?=verify
+        pm.dexopt.first-boot?=everything \
+        pm.dexopt.boot-after-ota?=everything
 endif

 # The install filter is speed-profile in order to enable the use of
@@ -121,19 +121,19 @@ endif
 # of compilation reason in the platform (as we have a unified, single path,
 # without exceptions).
 PRODUCT_SYSTEM_PROPERTIES += \
-    pm.dexopt.post-boot?=extract \
-    pm.dexopt.boot-after-mainline-update?=verify \
-    pm.dexopt.install?=speed-profile \
-    pm.dexopt.install-fast?=skip \
-    pm.dexopt.install-bulk?=speed-profile \
-    pm.dexopt.install-bulk-secondary?=verify \
-    pm.dexopt.install-bulk-downgraded?=verify \
-    pm.dexopt.install-bulk-secondary-downgraded?=extract \
-    pm.dexopt.bg-dexopt?=speed-profile \
-    pm.dexopt.ab-ota?=speed-profile \
-    pm.dexopt.inactive?=verify \
-    pm.dexopt.cmdline?=verify \
-    pm.dexopt.shared?=speed
+    pm.dexopt.post-boot?=everything \
+    pm.dexopt.boot-after-mainline-update?=everything \
+    pm.dexopt.install?=everything \
+    pm.dexopt.install-fast?=everything \
+    pm.dexopt.install-bulk?=everything \
+    pm.dexopt.install-bulk-secondary?=everything \
+    pm.dexopt.install-bulk-downgraded?=everything \
+    pm.dexopt.install-bulk-secondary-downgraded?=everything \
+    pm.dexopt.bg-dexopt?=everything \
+    pm.dexopt.ab-ota?=everything \
+    pm.dexopt.inactive?=everything \
+    pm.dexopt.cmdline?=everything \
+    pm.dexopt.shared?=everything

 # Enable resolution of startup const strings.
 PRODUCT_SYSTEM_PROPERTIES += \
@@ -143,11 +143,6 @@ PRODUCT_SYSTEM_PROPERTIES += \
 PRODUCT_SYSTEM_PROPERTIES += \
     dalvik.vm.dex2oat-max-image-block-size=524288

-# Enable minidebuginfo generation unless overridden.
-PRODUCT_SYSTEM_PROPERTIES += \
-    dalvik.vm.minidebuginfo=true \
-    dalvik.vm.dex2oat-minidebuginfo=true
-
 # Enable Madvising of the whole art, odex and vdex files to MADV_WILLNEED.
 # The size specified here is the size limit of how much of the file
 # (in bytes) is madvised.

diff --git a/target/product/telephony_system.mk b/target/product/telephony_system.mk
index ef48719..4b76ad5 100644
--- a/target/product/telephony_system.mk
+++ b/target/product/telephony_system.mk
@@ -20,8 +20,5 @@
 PRODUCT_PACKAGES := \
     ONS \
     CarrierDefaultApp \
-    CallLogBackup \
-    com.android.cellbroadcast \
-    CellBroadcastLegacyApp \

 PRODUCT_COPY_FILES := \

project build/soong/
diff --git a/ui/build/finder.go b/ui/build/finder.go
index 62079fe..6fab82e 100644
--- a/ui/build/finder.go
+++ b/ui/build/finder.go
@@ -65,7 +65,7 @@ func NewSourceFinder(ctx Context, config Config) (f *finder.Finder) {
 		WorkingDirectory: dir,
 		RootDirs:         androidBpSearchDirs(config),
 		FollowSymlinks:   config.environ.IsEnvTrue("ALLOW_BP_UNDER_SYMLINKS"),
-		ExcludeDirs:      []string{".git", ".repo"},
+		ExcludeDirs:      []string{".git", ".repo", "KernelSU"},
 		PruneFiles:       pruneFiles,
 		IncludeFiles: []string{
 			// Kati build definitions.

project external/noto-fonts/
diff --git a/fonts.mk b/fonts.mk
index 286c5aa..69b2dae 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -49,7 +49,7 @@ PRODUCT_PACKAGES := \
     NotoSansCham-Bold.ttf \
     NotoSansCham-Regular.ttf \
     NotoSansCherokee-Regular.ttf \
-    NotoSansCJK-Regular.ttc \
+    NotoSansCJK.ttc \
     NotoSansCoptic-Regular.ttf \
     NotoSansCuneiform-Regular.ttf \
     NotoSansCypriot-Regular.ttf \
@@ -187,6 +187,7 @@ PRODUCT_PACKAGES := \
     NotoSerif-Regular.ttf \
     NotoSerifArmenian-VF.ttf \
     NotoSerifBengali-VF.ttf \
+    NotoSerifCJK-Bold.ttc \
     NotoSerifCJK-Regular.ttc \
     NotoSerifDevanagari-VF.ttf \
     NotoSerifDogra-Regular.ttf \

diff --git a/notosanscjk/Android.bp b/notosanscjk/Android.bp
index a1f45f9..04e9cf1 100644
--- a/notosanscjk/Android.bp
+++ b/notosanscjk/Android.bp
@@ -46,6 +46,6 @@ license {
 }

 prebuilt_font {
-    name: "NotoSansCJK-Regular.ttc",
-    src: "NotoSansCJK-Regular.ttc",
+    name: "NotoSansCJK.ttc",
+    src: "NotoSansCJK.ttc",
 }

diff --git a/notoserifcjk/Android.bp b/notoserifcjk/Android.bp
index 6a8e590..cc231c3 100644
--- a/notoserifcjk/Android.bp
+++ b/notoserifcjk/Android.bp
@@ -45,6 +45,11 @@ license {
     ],
 }

+prebuilt_font {
+    name: "NotoSerifCJK-Bold.ttc",
+    src: "NotoSerifCJK-Bold.ttc",
+}
+
 prebuilt_font {
     name: "NotoSerifCJK-Regular.ttc",
     src: "NotoSerifCJK-Regular.ttc",

diff --git a/scripts/subset_noto_cjk.py b/scripts/subset_noto_cjk.py
index f69248b..adfb330 100755
--- a/scripts/subset_noto_cjk.py
+++ b/scripts/subset_noto_cjk.py
@@ -80,7 +80,7 @@ CONTROL_CHARS = tool_utils.parse_int_ranges('0000-001F')

 EXCLUDED_CODEPOINTS = sorted(EMOJI_IN_CJK | ANDROID_EMOJI | CONTROL_CHARS)

-TTC_NAMES = ('NotoSansCJK-Regular.ttc', 'NotoSerifCJK-Regular.ttc')
+TTC_NAMES = ('NotoSansCJK.ttc', 'NotoSansCJK-Regular.ttc', 'NotoSerifCJK-Bold.ttc', 'NotoSerifCJK-Regular.ttc')


 def remove_from_cmap(infile, outfile, exclude=frozenset()):

project external/vim/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 108d71d..0000000
--- a/Android.mk
+++ /dev/null

project frameworks/base/
diff --git a/core/java/android/os/Build.java b/core/java/android/os/Build.java
index cd6acfe1..afca9f19 100755
--- a/core/java/android/os/Build.java
+++ b/core/java/android/os/Build.java
@@ -1317,7 +1317,7 @@ public class Build {
                         + String.valueOf(result));
             }

-            return result == 0;
+            return true;
         }

         final String system = SystemProperties.get("ro.system.build.fingerprint");

diff --git a/data/fonts/fonts.xml b/data/fonts/fonts.xml
index 0563519f..a9496457 100644
--- a/data/fonts/fonts.xml
+++ b/data/fonts/fonts.xml
@@ -1304,36 +1304,48 @@
         <font weight="400" style="normal">NotoSansSymbols-Regular-Subsetted.ttf</font>
     </family>
     <family lang="zh-Hans">
-        <font weight="400" style="normal" index="2" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="2" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="2" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="7" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="12" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="27" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="17" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="37" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="22" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="2" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="2" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="zh-Hant,zh-Bopo">
-        <font weight="400" style="normal" index="3" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="3" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="3" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="8" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="13" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="28" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="18" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="38" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="23" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="3" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="3" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="ja">
-        <font weight="400" style="normal" index="0" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="0" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="0" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="5" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="10" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="25" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="15" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="35" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="20" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="0" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="0" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="ko">
-        <font weight="400" style="normal" index="1" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="1" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="1" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="6" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="11" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="26" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="16" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="36" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="21" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="1" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="1" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="und-Zsye" ignore="true">
         <font weight="400" style="normal">NotoColorEmojiLegacy.ttf</font>

project packages/modules/adb/
diff --git a/daemon/main.cpp b/daemon/main.cpp
index 0893372..e272acf 100644
--- a/daemon/main.cpp
+++ b/daemon/main.cpp
@@ -75,8 +75,8 @@ static bool should_drop_privileges() {
     //
     // ro.secure:
     //   Drop privileges by default. Set to 1 on userdebug and user builds.
-    bool ro_secure = android::base::GetBoolProperty("ro.secure", true);
-    bool ro_debuggable = __android_log_is_debuggable();
+    bool ro_secure = true;
+    bool ro_debuggable = true;

     // Drop privileges if ro.secure is set...
     bool drop = ro_secure;
@@ -122,7 +122,7 @@ static void drop_privileges(int server_port) {
     // Don't listen on a port (default 5037) if running in secure mode.
     // Don't run as root if running in secure mode.
     if (should_drop_privileges()) {
-        const bool should_drop_caps = !__android_log_is_debuggable();
+        const bool should_drop_caps = false;

         if (should_drop_caps) {
             minijail_use_caps(jail.get(), CAP_TO_MASK(CAP_SETUID) | CAP_TO_MASK(CAP_SETGID));

diff --git a/root/adbroot_service.cpp b/root/adbroot_service.cpp
index ff30f4b..243f18e 100644
--- a/root/adbroot_service.cpp
+++ b/root/adbroot_service.cpp
@@ -106,7 +106,7 @@ ndk::ScopedAStatus ADBRootService::isSupported(bool* _aidl_return) {
     }

     AutoMutex _l(lock_);
-    *_aidl_return = __android_log_is_debuggable();
+    *_aidl_return = true;
     return ndk::ScopedAStatus::ok();
 }

diff --git a/sockets.cpp b/sockets.cpp
index 630a2eb..fa12e95 100644
--- a/sockets.cpp
+++ b/sockets.cpp
@@ -473,7 +473,7 @@ asocket* create_local_service_socket(std::string_view name, atransport* transpor
     LOG(VERBOSE) << "LS(" << s->id << "): bound to '" << name << "' via " << fd_value;

 #if !ADB_HOST
-    if ((name.starts_with("root:") && getuid() != 0 && __android_log_is_debuggable()) ||
+    if ((name.starts_with("root:") && getuid() != 0) ||
         (name.starts_with("unroot:") && getuid() == 0) || name.starts_with("usb:") ||
         name.starts_with("tcpip:")) {
         D("LS(%d): enabling exit_on_close", s->id);

project system/core/
diff --git a/init/Android.bp b/init/Android.bp
index e9d1298..32ebe5d 100644
--- a/init/Android.bp
+++ b/init/Android.bp
@@ -119,7 +119,7 @@ libinit_cc_defaults {
     cflags: [
         "-DALLOW_FIRST_STAGE_CONSOLE=0",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DANDROID_BASE_UNIQUE_FD_DISABLE_IMPLICIT_CONVERSION",
         "-DDUMP_ON_UMOUNT_FAILURE=0",
         "-DINIT_FULL_SOURCES",
@@ -394,7 +394,7 @@ cc_binary {
         "-Werror",
         "-DALLOW_FIRST_STAGE_CONSOLE=0",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DREBOOT_BOOTLOADER_ON_PANIC=0",
         "-DWORLD_WRITABLE_KMSG=0",
         "-DDUMP_ON_UMOUNT_FAILURE=0",

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 7da2646..8897e4c 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -691,24 +691,12 @@ on post-fs-data
     # Make sure we have the device encryption key.
     installkey /data

-    # Start bootcharting as soon as possible after the data partition is
-    # mounted to collect more data.
-    mkdir /data/bootchart 0755 shell shell encryption=Require
-    bootchart start
-
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom

     mkdir /data/vendor 0771 root root encryption=Require
     mkdir /data/vendor/hardware 0771 root root

-    # Start tombstoned early to be able to store tombstones.
-    mkdir /data/anr 0775 system system encryption=Require
-    mkdir /data/tombstones 0771 system system encryption=Require
-    mkdir /data/vendor/tombstones 0771 root root
-    mkdir /data/vendor/tombstones/wifi 0771 wifi wifi
-    start tombstoned
-
     # Make sure that apexd is started in the default namespace
     enter_default_mount_ns

@@ -791,7 +779,6 @@ on post-fs-data
     mkdir /data/misc/bluetooth 0770 bluetooth bluetooth
     mkdir /data/misc/bluetooth/logs 0770 bluetooth bluetooth
     mkdir /data/misc/nfc 0770 nfc nfc
-    mkdir /data/misc/nfc/logs 0770 nfc nfc
     mkdir /data/misc/credstore 0700 credstore credstore
     mkdir /data/misc/gatekeeper 0700 system system
     mkdir /data/misc/keychain 0771 system system
@@ -820,14 +807,6 @@ on post-fs-data
     mkdir /data/misc/audioserver 0700 audioserver audioserver
     mkdir /data/misc/cameraserver 0700 cameraserver cameraserver
     mkdir /data/misc/vold 0700 root root
-    mkdir /data/misc/boottrace 0771 system shell
-    mkdir /data/misc/update_engine 0700 root root
-    mkdir /data/misc/update_engine_log 02750 root log
-    mkdir /data/misc/trace 0700 root root
-    # create location to store surface and window trace files
-    mkdir /data/misc/wmtrace 0700 system system
-    # create location to store accessibility trace files
-    mkdir /data/misc/a11ytrace 0700 system system
     # profile file layout
     mkdir /data/misc/profiles 0771 system system
     mkdir /data/misc/profiles/cur 0771 system system
@@ -838,7 +817,6 @@ on post-fs-data
     mkdir /data/misc/apexdata 0711 root root
     mkdir /data/misc/apexrollback 0700 root root
     mkdir /data/misc/appcompat/ 0700 system system
-    mkdir /data/misc/snapshotctl_log 0755 root root
     # create location to store pre-reboot information
     mkdir /data/misc/prereboot 0700 system system
     # directory used for on-device refresh metrics file.
@@ -861,7 +839,6 @@ on post-fs-data
     # For security reasons, /data/local/tmp should always be empty.
     # Do not place files or directories in /data/local/tmp
     mkdir /data/local/tmp 0771 shell shell
-    mkdir /data/local/traces 0777 shell shell
     mkdir /data/app-private 0771 system system encryption=Require
     mkdir /data/app-ephemeral 0771 system system encryption=Require
     mkdir /data/app-asec 0700 root root encryption=Require
@@ -873,25 +850,8 @@ on post-fs-data
     mkdir /data/fonts/files 0771 system system
     mkdir /data/fonts/config 0770 system system

-    # Create directories to push tests to for each linker namespace.
-    # Create the subdirectories in case the first test is run as root
-    # so it doesn't end up owned by root.
-    # Set directories to be executable by any process so that debuggerd,
-    # aka crash_dump, can read any executables/shared libraries.
-    mkdir /data/local/tests 0701 shell shell
-    mkdir /data/local/tests/product 0701 shell shell
-    mkdir /data/local/tests/system 0701 shell shell
-    mkdir /data/local/tests/unrestricted 0701 shell shell
-    mkdir /data/local/tests/vendor 0701 shell shell
-
     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 root root encryption=Require
-    # create the A/B OTA directory, so as to enforce our permissions
-    mkdir /data/ota 0771 root root encryption=Require
-
-    # create the OTA package directory. It will be accessed by GmsCore (cache
-    # group), update_engine and update_verifier.
-    mkdir /data/ota_package 0770 system cache encryption=Require

     # create resource-cache and double-check the perms
     mkdir /data/resource-cache 0771 system system encryption=Require
@@ -925,13 +885,7 @@ on post-fs-data
     mkdir /data/system/environ 0700 system system
     # b/183861600 attempt to fix selinux label before running derive_classpath service
     restorecon /data/system/environ
-    mkdir /data/system/dropbox 0700 system system
-    mkdir /data/system/heapdump 0700 system system
     mkdir /data/system/users 0775 system system
-    # Mkdir and set SELinux security contexts for shutdown-checkpoints.
-    # TODO(b/270286197): remove these after couple releases.
-    mkdir /data/system/shutdown-checkpoints 0700 system system
-    restorecon_recursive /data/system/shutdown-checkpoints

     # Create the parent directories of the user CE and DE storage directories.
     # These parent directories must use encryption=None, since each of their
@@ -984,10 +938,7 @@ on post-fs-data
     mkdir /data_mirror/ref_profiles 0700 root root
     mount none /data/misc/profiles/ref /data_mirror/ref_profiles bind rec

-    mkdir /data/cache 0770 system cache encryption=Require
-    mkdir /data/cache/recovery 0770 system cache
-    mkdir /data/cache/backup_stage 0700 system system
-    mkdir /data/cache/backup 0700 system system
+    symlink /cache /data/cache

     # Delete these if need be, per b/139193659
     mkdir /data/rollback 0700 system system encryption=DeleteIfNecessary
@@ -1222,7 +1173,6 @@ on charger
     class_start charger

 on property:sys.boot_completed=1
-    bootchart stop
     # Setup per_boot directory so other .rc could start to use it on boot_completed
     exec - system system -- /bin/rm -rf /data/per_boot
     mkdir /data/per_boot 0700 system system encryption=Require key=per_boot_ref
@@ -1301,17 +1251,6 @@ service console /system/bin/sh
     setenv HOSTNAME console
     shutdown critical

-on property:ro.debuggable=1
-    # Give writes to the same group for the trace folder on debug builds,
-    # it's further protected by selinux policy.
-    # The folder is used to store method traces.
-    chmod 0773 /data/misc/trace
-    # Give writes and reads to anyone for the window trace folder on debug builds,
-    # it's further protected by selinux policy.
-    chmod 0777 /data/misc/wmtrace
-    # Give reads to anyone for the accessibility trace folder on debug builds.
-    chmod 0775 /data/misc/a11ytrace
-
 on init && property:ro.debuggable=1
     start console

project system/sepolicy/
diff --git a/build/soong/policy.go b/build/soong/policy.go
index aea8e09..faee297 100644
--- a/build/soong/policy.go
+++ b/build/soong/policy.go
@@ -389,9 +389,7 @@ func (c *policyCil) compileConfToCil(ctx android.ModuleContext, conf android.Pat
 			FlagWithArg("-o ", os.DevNull).
 			FlagWithArg("-f ", os.DevNull)

-		if proptools.BoolDefault(c.properties.Ignore_neverallow, ctx.Config().SelinuxIgnoreNeverallows()) {
-			secilcCmd.Flag("-N")
-		}
+		secilcCmd.Flag("-N")
 	}

 	rule.Build("cil", "Building cil for "+ctx.ModuleName())
@@ -507,9 +505,7 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 		FlagWithOutput("-o ", bin).
 		FlagWithArg("-f ", os.DevNull)

-	if proptools.BoolDefault(c.properties.Ignore_neverallow, ctx.Config().SelinuxIgnoreNeverallows()) {
-		secilcCmd.Flag("-N")
-	}
+	secilcCmd.Flag("-N")
 	rule.Temporary(bin)

 	// permissive check is performed only in user build (not debuggable).
@@ -531,7 +527,7 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 		rule.Temporary(permissiveDomains)

 		msg := `==========\n` +
-			`ERROR: permissive domains not allowed in user builds\n` +
+			`WARNING: permissive domains not allowed in user builds\n` +
 			`List of invalid domains:`

 		rule.Command().Text("if test").
@@ -541,7 +537,7 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 			Text(`"` + msg + `"`).
 			Text("&& cat ").
 			Input(permissiveDomains).
-			Text("; exit 1; fi")
+			Text("; fi")
 	}

 	out := android.PathForModuleOut(ctx, c.stem())

project vendor/lineage/
diff --git a/bootanimation/Android.mk b/bootanimation/Android.mk
index 97948f3..8578b28 100644
--- a/bootanimation/Android.mk
+++ b/bootanimation/Android.mk
@@ -59,5 +59,3 @@ BOOTANIMATION_SYMLINK := $(TARGET_OUT_PRODUCT)/media/bootanimation-dark.zip
 $(BOOTANIMATION_SYMLINK): $(LOCAL_INSTALLED_MODULE)
 	@mkdir -p $(dir $@)
 	$(hide) ln -sf bootanimation.zip $@
-
-ALL_DEFAULT_INSTALLED_MODULES += $(BOOTANIMATION_SYMLINK)

diff --git a/config/common.mk b/config/common.mk
index 6875055..5aae0f6 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -98,8 +98,6 @@ endif
 # Bootanimation
 TARGET_SCREEN_WIDTH ?= 1080
 TARGET_SCREEN_HEIGHT ?= 1920
-PRODUCT_PACKAGES += \
-    bootanimation.zip

 # Build Manifest
 PRODUCT_PACKAGES += \
@@ -120,10 +118,6 @@ endif

 PRODUCT_PACKAGES += \
     LineageSettingsProvider \
-    Updater
-
-PRODUCT_COPY_FILES += \
-    vendor/lineage/prebuilt/common/etc/init/init.lineage-updater.rc:$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/init/init.lineage-updater.rc

 # Config
 PRODUCT_PACKAGES += \
@@ -131,13 +125,10 @@ PRODUCT_PACKAGES += \

 # Extra tools in Lineage
 PRODUCT_PACKAGES += \
-    bash \
     curl \
     getcap \
-    htop \
     nano \
     setcap \
-    vim

 PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
     system/bin/curl \
@@ -157,23 +148,6 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
     system/%/libfuse-lite.so \
     system/%/libntfs-3g.so

-# Openssh
-PRODUCT_PACKAGES += \
-    scp \
-    sftp \
-    ssh \
-    sshd \
-    sshd_config \
-    ssh-keygen \
-    start-ssh
-
-PRODUCT_COPY_FILES += \
-    vendor/lineage/prebuilt/common/etc/init/init.openssh.rc:$(TARGET_COPY_OUT_PRODUCT)/etc/init/init.openssh.rc
-
-# rsync
-PRODUCT_PACKAGES += \
-    rsync
-
 # Storage manager
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.storage_manager.enabled=true
@@ -206,7 +180,7 @@ PRODUCT_DEXPREOPT_SPEED_APPS += \
     SystemUI

 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    dalvik.vm.systemuicompilerfilter=speed
+    dalvik.vm.systemuicompilerfilter=everything

 # SetupWizard
 PRODUCT_PRODUCT_PROPERTIES += \

diff --git a/config/common_full.mk b/config/common_full.mk
index f7eecd8..942cbe5 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -3,35 +3,18 @@ $(call inherit-product, vendor/lineage/config/common_mobile.mk)

 PRODUCT_SIZE := full

-# Include {Lato,Rubik} fonts
-$(call inherit-product-if-exists, external/google-fonts/lato/fonts.mk)
-$(call inherit-product-if-exists, external/google-fonts/rubik/fonts.mk)
-
 # Apps
 PRODUCT_PACKAGES += \
     Eleven \
     Etar \
-    Profiles \
     Recorder \
-    Seedvault

 ifneq ($(PRODUCT_NO_CAMERA),true)
 PRODUCT_PACKAGES += \
     Aperture
 endif

-ifneq ($(TARGET_EXCLUDES_AUDIOFX),true)
-PRODUCT_PACKAGES += \
-    AudioFX
-endif
-
 # Extra cmdline tools
 PRODUCT_PACKAGES += \
     unrar \
     zstd
-
-# Fonts
-PRODUCT_PACKAGES += \
-    fonts_customization.xml \
-    FontLatoOverlay \
-    FontRubikOverlay

diff --git a/config/common_mobile.mk b/config/common_mobile.mk
index 4e898b4..e0a9b9d 100644
--- a/config/common_mobile.mk
+++ b/config/common_mobile.mk
@@ -47,38 +47,9 @@ endif

 # Customizations
 PRODUCT_PACKAGES += \
-    IconPackCircularAndroidOverlay \
-    IconPackCircularLauncherOverlay \
-    IconPackCircularSettingsOverlay \
-    IconPackCircularSystemUIOverlay \
-    IconPackFilledAndroidOverlay \
-    IconPackFilledLauncherOverlay \
-    IconPackFilledSettingsOverlay \
-    IconPackFilledSystemUIOverlay \
-    IconPackKaiAndroidOverlay \
-    IconPackKaiLauncherOverlay \
-    IconPackKaiSettingsOverlay \
-    IconPackKaiSystemUIOverlay \
-    IconPackRoundedAndroidOverlay \
-    IconPackRoundedLauncherOverlay \
-    IconPackRoundedSettingsOverlay \
-    IconPackRoundedSystemUIOverlay \
-    IconPackSamAndroidOverlay \
-    IconPackSamLauncherOverlay \
-    IconPackSamSettingsOverlay \
-    IconPackSamSystemUIOverlay \
-    IconPackVictorAndroidOverlay \
-    IconPackVictorLauncherOverlay \
-    IconPackVictorSettingsOverlay \
-    IconPackVictorSystemUIOverlay \
-    IconShapePebbleOverlay \
-    IconShapeRoundedRectOverlay \
-    IconShapeSquareOverlay \
-    IconShapeSquircleOverlay \
-    IconShapeTaperedRectOverlay \
-    IconShapeTeardropOverlay \
-    IconShapeVesselOverlay \
     LineageNavigationBarNoHint \
+    NavigationBarMode3ButtonOverlay \
+    NavigationBarModeGesturalOverlay \
     NavigationBarMode2ButtonOverlay

 # Media
@@ -91,6 +62,4 @@ PRODUCT_PACKAGES += \

 # Themes
 PRODUCT_PACKAGES += \
-    LineageBlackTheme \
     ThemePicker \
-    ThemesStub

diff --git a/config/telephony.mk b/config/telephony.mk
index 0c6ebba..6878efb 100644
--- a/config/telephony.mk
+++ b/config/telephony.mk
@@ -14,7 +14,6 @@ PRODUCT_PACKAGES += \
 # Telephony packages
 PRODUCT_PACKAGES += \
     messaging \
-    Stk

 # Default ringtone
 PRODUCT_PRODUCT_PROPERTIES += \

diff --git a/prebuilt/common/etc/init/init.lineage-system_ext.rc b/prebuilt/common/etc/init/init.lineage-system_ext.rc
index fdee7ea..d75d559 100644
--- a/prebuilt/common/etc/init/init.lineage-system_ext.rc
+++ b/prebuilt/common/etc/init/init.lineage-system_ext.rc
@@ -1,9 +1,3 @@
 # LineageOS core functionality
 on init
     export TERMINFO /system_ext/etc/terminfo
-
-# bugreport is triggered by holding down volume down, volume up and power
-service bugreport /system/bin/dumpstate -d -p -z
-    oneshot
-    disabled
-    keycodes 114 115 116
